src/tools/lrucache.h

```diff
@@ -138,12 +138,18 @@ class lru_cache {
     return _cache_items_map.size();
   }
 
+  size_t set_max_size(size_t new_size) {
```

Comment:
I know that this class is a hybrid of a snake with a camel, but I wonder if your choice of the style here was deliberate :)

Now getting serious. If the cache's current size exceeds the new value of max size, shouldn't it be truncated immediately? Though, a deeper question is - do we really need a dynamic cache size at all (i.e. is linking the cache size to the actual amount of data a good idea)?